system/Config/Services.php

```diff
@@ -220,34 +221,33 @@ public static function email($config = null, bool $getShared = true)
 		{
 			return static::getSharedInstance('email', $config);
 		}
-		if (empty($config))
+
+		if (! is_array($config) && ! $config instanceof EmailConfig)
```

Comment:
I was trying to target here forcing the right type for `$config` since it is not type hinted like the others. the empty check will fail if I erroneously passed a complete string or values other than the intended `array` type or instance of `Config\Email`.

Perhaps I can add the empty check before the other two checks for completeness.

Reply:
How about this? Should cover all the cases.
```suggestion
		if ((is_array($config) && empty($config)) || ! is_array($config) && ! $config instanceof EmailConfig)
```

Reply:
I was actually thinking of this:
```php
if (empty($config) || ! is_array($config) && ! $config instanceof EmailConfig)
```

Same logic but eliminated use of two is_array checks.